contracts/test/PoolFDTLosses.t.sol

```diff
@@ -0,0 +1,314 @@
+
+        // Time warp to withdraw
+        hevm.warp(block.timestamp + pool.lockupPeriod() + 1);
+
+        // Fails to transfer if the losses are > 0.
+        assertTrue(
+            !leo.try_transfer(address(pool), address(leo), 1),
+            "Should not allowed to transfer because losses are > 0"
+        );
+
+        // Lex is withdrawing funds from the Pool with losses.
+        assertTrue(leo.try_intendToWithdraw(address(pool)), "Failed to intend to withdraw");
+
+        // Time warp to withdraw window
+        hevm.warp(block.timestamp + globals.lpCooldownPeriod() + 1);
+
+        // Before withdrawing funds.
+        uint256 old_lex_bal = usdc.balanceOf(address(leo));
+
+        assertTrue(leo.try_withdraw(address(pool), depositAmt), "Failed to withdraw");
+
+        assertEq(
+            (usdc.balanceOf(address(leo)) - old_lex_bal) / USD,
+            (depositAmt - poolLosses) / USD,
+            "New balance should count the losses"
+        );
+    }
+
+    function test_multipleLpBearingLossesWithMultiplePools(
+        uint256 apr,
+        uint256 index,
+        uint256 numPayments,
+        uint256 requestAmount,
+        uint256 leoDepositAmt,
+        uint256 lexDepositAmt
+    ) public {
+        uint256 minDepositAmountForPool1 = PoolLib.getSwapOutValueLocker(address(bPool), USDC, address(pool.stakeLocker()));
+        uint256 minDepositAmountForPool2 = PoolLib.getSwapOutValueLocker(address(bPool), USDC, address(pool2.stakeLocker()));
+
+        // Create Loan and a deposit amount from arguments
+        (Loan custom_loan, uint256 totalDepositAmt) = getLoanAndDepositAmount(apr, index, numPayments, requestAmount, 0, minDepositAmountForPool1 + minDepositAmountForPool2);
+
+        uint256 lizDepositAmt;
+        (leoDepositAmt, lizDepositAmt) = getSplitDepositAmounts(leoDepositAmt, (minDepositAmountForPool1 * totalDepositAmt) / (minDepositAmountForPool1 + minDepositAmountForPool2));
+
+        uint256 leeDepositAmt;
+        (lexDepositAmt, leeDepositAmt) = getSplitDepositAmounts(lexDepositAmt, totalDepositAmt - leoDepositAmt - lizDepositAmt);
+
+        // LPs is putting up the funds into the Pool.
+        noAssert_mintFundsAndDepositIntoPool(leo, pool,  leoDepositAmt, leoDepositAmt);
+        noAssert_mintFundsAndDepositIntoPool(liz, pool,  lizDepositAmt, lizDepositAmt);
+        noAssert_mintFundsAndDepositIntoPool(lex, pool2, lexDepositAmt, lexDepositAmt);
+        noAssert_mintFundsAndDepositIntoPool(lee, pool2, leeDepositAmt, leeDepositAmt);
+
+        // Fund the loan by both pool delegates.
+        pat.fundLoan(address(pool),  address(custom_loan), address(dlFactory), usdc.balanceOf(pool.liquidityLocker()));
+        pam.fundLoan(address(pool2), address(custom_loan), address(dlFactory), usdc.balanceOf(pool2.liquidityLocker()));
+
+        // Drawdown of the loan
+        drawdown(custom_loan, bob, totalDepositAmt);
+
+        /// Time warp to make a payment.
```

Comment:
//

Reply:
// Time warp and make payment
and then include the `doPartialLoanPayment(custom_loan, bob);` line below